apis/goharbor.io/v1beta1/chartmuseum_conversion.go

```diff
@@ -0,0 +1,37 @@
+package v1beta1
+
+import (
+	"github.com/goharbor/harbor-operator/apis/goharbor.io/v1alpha3"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/conversion"
+)
+
+func (c *ChartMuseum) ConvertTo(dstRaw conversion.Hub) error {
+	src, err := runtime.DefaultUnstructuredConverter.ToUnstructured(c)
+	if err != nil {
+		return err
+	}
+
+	if err = runtime.DefaultUnstructuredConverter.FromUnstructured(src, dstRaw); err != nil {
+		return err
+	}
+
+	dstRaw.(*v1alpha3.ChartMuseum).APIVersion = v1alpha3.GroupVersion.String()
+
+	return nil
+}
+
+func (c *ChartMuseum) ConvertFrom(srcRaw conversion.Hub) error {
+	src, err := runtime.DefaultUnstructuredConverter.ToUnstructured(srcRaw)
+	if err != nil {
+		return err
+	}
+
+	if err = runtime.DefaultUnstructuredConverter.FromUnstructured(src, c); err != nil {
+		return err
+	}
+
+	c.APIVersion = GroupVersion.String()
+
+	return nil
```

Comment:
```suggestion
	gvk := c.GetObjectKind().GroupVersionKind()
	defer func() {
		c.GetObjectKind().SetGroupVersionKind(gvk)
	}()

	src, err := runtime.DefaultUnstructuredConverter.ToUnstructured(srcRaw)
	if err != nil {
		return err
	}

	return runtime.DefaultUnstructuredConverter.FromUnstructured(src, c)
```

Reply:
It seems to be generic enough to be considered for integration?

Reply:
Wonderful！I think it can extract a method.

`func ConvertViaUnstructuredCopy(src, dest runtime.Object) error {
	gvk := dest.GetObjectKind().GroupVersionKind()
	defer func() {
		dest.GetObjectKind().SetGroupVersionKind(gvk)
	}()

	srcRaw, err := runtime.DefaultUnstructuredConverter.ToUnstructured(src)
	if err != nil {
		return err
	}

	return runtime.DefaultUnstructuredConverter.FromUnstructured(srcRaw, dest)
}`